### PR TITLE
fix: Limit workflows to 128KB and return a friendly error message

### PR DESCRIPTION
**Pull request by blkperl**
Signed-off-by: William Van Hevelingen <william.vanhevelingen@acquia.com>

Part of: #7586

### CODE REVIEW

File: test/e2e/expectedfailures/large-workflow.yaml [hunk 1]
```diff
@@ -3,15 +3,1572 @@ kind: Workflow
 metadata:
   generateName: large-workflow-
 spec:
-  entrypoint: large-workflow
```
Comment (blkperl) [inline test/e2e/expectedfailures/large-workflow.yaml:6]: @jessesuen added this file in #913 but it doesn't look like the e2e tests use it.

File: workflow/controller/workflowpod.go [hunk 2]
```diff
@@ -317,9 +317,16 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 		pod.Spec.InitContainers[i] = c
 	}
 
+	envVarTemplateValue := wfv1.MustMarshallJSON(tmpl)
+	templateSize := len(envVarTemplateValue)
+	if templateSize > 128000 {
```
Comment (blkperl) [inline workflow/controller/workflowpod.go:322]: ARG_MAX is 131KB so we could allow a slightly higher number but there are other environment variables being added. 

```
docker run --entrypoint ""  -it quay.io/argoproj/argoexec:latest sh                       
/ # getconf ARG_MAX
131072
```

Comment (blkperl) [reply workflow/controller/workflowpod.go:322]: We could loop over all the `envVars` and add them in a separate error message.

**Review by alexec — APPROVED** [review]

File: workflow/controller/workflowpod.go (hunk 2 above)
Comment (blkperl) [reply workflow/controller/workflowpod.go:322]: @alexec is there any special way to note this as a breaking change and where would be the best place to document the new limit?

Comment (alexec) [reply workflow/controller/workflowpod.go:322]: I thought we were going to fix this by splitting the single ARG (max 130k) into two args (make 260k) which would fix the issue.

Comment (alexec) [reply workflow/controller/workflowpod.go:322]: Docs are still good mind you, so I approve of this.

**Pull request merged**